crates/net/network/src/eth_requests.rs

```diff
@@ -201,26 +201,28 @@ where
         response: oneshot::Sender<RequestResult<Receipts>>,
     ) {
         let mut receipts = Vec::new();
-
-        let mut total_bytes = APPROX_RECEIPT_SIZE;
+        let mut total_len = 0;
 
         for hash in request.0 {
             if let Some(receipts_by_block) =
                 self.client.receipts_by_block(BlockHashOrNumber::Hash(hash)).unwrap_or_default()
             {
-                receipts.push(
-                    receipts_by_block
-                        .into_iter()
-                        .map(|receipt| receipt.with_bloom())
-                        .collect::<Vec<_>>(),
-                );
+                let block_receipts = receipts_by_block
+                    .into_iter()
+                    .map(|receipt| receipt.with_bloom())
+                    .collect::<Vec<_>>();
 
-                total_bytes += APPROX_RECEIPT_SIZE;
+                let mut buf = BytesMut::new();
+                block_receipts.encode(&mut buf);
 
-                if total_bytes > SOFT_RESPONSE_LIMIT {
+                total_len += buf.len();
+
+                if total_len > SOFT_RESPONSE_LIMIT {
                     break
                 }
 
+                receipts.push(block_receipts);
```

Comment:
we actually want the push before the check, to ensure we always send at least one, this is a bug on main